core/src/main/java/com/novoda/merlin/MerlinsBeard.java

```diff
@@ -44,16 +48,34 @@ private NetworkInfo getNetworkInfo() {
     }
 
     /**
-     * Provides a boolean representing whether a wifi network connection has been established.
+     * Provides a boolean representing whether a Wi-Fi network connection has been established.
      *
      * NOTE: Therefore available does not necessarily mean that an internet connection
      * is available.
      *
-     * @return boolean true if a wifi network connection is available.
+     * @return boolean true if a Wi-Fi network connection is available.
      */
     public boolean isConnectedToWifi() {
         NetworkInfo networkInfo = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
         return networkInfo != null && networkInfo.isConnected();
     }
 
+
+    /**
+     * Provides a boolean representing whether a mobile network connection has been established.
+     * NOTE: Therefore available does not necessarily mean that an internet connection
+     * is available.
+     *
+     * @return boolean true if a mobile network connection is available.
+     */
+    public boolean isConnectedToMobileNetwork() {
+        int simState = telephonyManager.getSimState();
+        if (simState == TelephonyManager.SIM_STATE_READY) {
```

Comment:
@blundell not sure how to phrase it differently. I would assume that the two methods isConnected and isConnectedToMobileNetwork have a similar semantics. However, it looks to me that they don't because the first is just inspecting the network info, the second is inspecting the network info and the sim state. I don't have the knowledge about network connectivity, maybe isConnectedToMobileNetwork should be renamed to isReadyAndConnectedToMobileNetwork or adding something like `tm.getDataState() == TelephonyManager.DATA_CONNECTED` would make it clear/useful that the method is looking into more than the network info.